File: server/internal/models/course.go
package models

import "strings"

// Ders
type Course struct {
	Code           string  `json:"code" db:"course_code"`
	DepartmentID   int     `json:"department_id" db:"department_id"`
	DepartmentGUID string  `json:"department_guid,omitempty"`
	Name           string  `json:"name" db:"course_name"`
	Credit         float64 `json:"credit" db:"credit"`
	ECTS           float64 `json:"ects" db:"ects"`
	IsMandatory    bool    `json:"is_mandatory" db:"is_mandatory"`
	Theory         int     `json:"theory" db:"theory_hours"`
	Practice       int     `json:"practice" db:"practice_hours"`
	Lab            int     `json:"lab" db:"lab_hours"`
	Semester       string  `json:"semester" db:"semester"`
	LinkID         string  `json:"link_id" db:"link_id"`
	UnitID         string  `json:"unit_id" db:"unit_id"`

	//Türetilmiş (EBS'den gelmiyor)
	Year      int  `json:"year" db:"year"`
	IsRemoved bool `json:"is_removed" db:"is_removed"`
}

// Ders Detayları
type CourseDetail struct {
	BaseInfo   Course   `json:"base_info"`
	Instructor string   `json:"instructor" db:"instructor"`
	Language   string   `json:"language" db:"language"`
	Aim        string   `json:"aim" db:"aim"`
	Content    string   `json:"content" db:"content"`
	Resources  string   `json:"resources" db:"resources"`
	Outcomes   []string `json:"outcomes" db:"outcomes"`
}

func (d *CourseDetail) HasContent() bool {
	return len(strings.TrimSpace(d.Aim)) > 0 ||
		len(strings.TrimSpace(d.Content)) > 0 ||
		len(d.Outcomes) > 0
}

File: iuc_companion/lib/data/models/course.dart
import 'package:floor/floor.dart';

// Changed: Added primaryKeys list to include departmentGuid
@Entity(primaryKeys: ['code', 'departmentGuid'])
class Course {
  // Removed @primaryKey annotation from here
  final String code;
  final int departmentId;
  final String departmentGuid; // Changed: Made non-nullable
  final String name;
  final double credit;
  final double ects;
  final bool isMandatory;
  final int theory;
  final int practice;
  final int lab;
  final String semester;
  final String linkId;
  final String unitId;
  final int? year;
  final bool isRemoved;

  Course({
    required this.code,
    required this.departmentId,
    required this.departmentGuid, // Required now
    required this.name,
    required this.credit,
    required this.ects,
    required this.isMandatory,
    required this.theory,
    required this.practice,
    required this.lab,
    required this.semester,
    required this.linkId,
    required this.unitId,
    this.year,
    required this.isRemoved,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      code: json['code'] ?? '',
      departmentId: json['department_id'] ?? 0,
      departmentGuid: json['department_guid'] ?? '',
      name: json['name'] ?? '',
      credit: (json['credit'] as num?)?.toDouble() ?? 0.0,
      ects: (json['ects'] as num?)?.toDouble() ?? 0.0,
      isMandatory: json['is_mandatory'] ?? false,
      theory: json['theory'] ?? 0,
      practice: json['practice'] ?? 0,
      lab: json['lab'] ?? 0,
      semester: json['semester'] ?? '',
      linkId: json['link_id'] ?? '',
      unitId: json['unit_id'] ?? '',
      year: json['year'],
      isRemoved: json['is_removed'] ?? false,
    );
  }
}